hornetq-core-client/src/main/java/org/hornetq/core/client/impl/ClientConsumerImpl.java

```diff
@@ -699,7 +699,7 @@ private void handleCompressedMessage(final SessionReceiveMessage message) throws
     */
    private File createLargeMessageTempFile(long messageId) throws IOException
    {
-      Set<PosixFilePermission> permissions = PosixFilePermissions.fromString("rw");
+      Set<PosixFilePermission> permissions = PosixFilePermissions.fromString("rw-------");
```

Comment:
Did you test this on Windows as iirc setting permissions there is a bit different.

Reply:
Also the permission set should be static and final

Reply:
I think
         largeMessageCache.setReadable(false);
         largeMessageCache.setWritable(false);
         largeMessageCache.setReadable(true, true);
         largeMessageCache.setWritable(true, true);
would be better

Reply:
Let me try this on Windows.  If we can specify permissions on the operation that triggers file creation I think that would be clearer and potentially less IO intensive.

Reply:
Ah, I think we may be overthinking this.
NIO's `Files.createTempFile` on OSX creates files with user-only read/write permissions.  I'm a bit unsure of how to test for the presence of group permissions on Windows, as the security model is quite different.

Still trying to come up with a solution that doesn't need platform-specific considerations and doesn't need additional permissions settings, it should be doable.

Reply:
Changed temporary file creation to create a directory on demand, set permissions on that directory on creation, and create temporary files in that directory.  I wrote down a bunch of details in the commit message for that update, feel free to take a look.